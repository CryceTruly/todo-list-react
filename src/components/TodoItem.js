import React from 'react'

function TodoItem(props) {
  return (
    <li  className='list-group-item'>
    <input className="toggle mr-4" type="checkbox" checked={props.isCompleted}  onChange={()=>props.handleToogle(props.id)}/>
   
   <strong>{props.id}#</strong> <span style={{textDecoration:props.isCompleted?'line-through':''}}><label>{props.name}</label></span>
   <button className='btn btn-danger btn-sm float-right' onClick={()=>props.handleRemove(props.id)}>x</button>
  
  </li>
  )
}

export default TodoItem;