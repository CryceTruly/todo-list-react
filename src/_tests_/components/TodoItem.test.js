import React from "react";
import {shallow } from "enzyme";
import  TodoItem  from "../../components/TodoItem";
describe("TodoApp", () => {

  const props = {
    handleRemove:jest.fn(),handleToogle:jest.fn(),
    todos:[{
        id:1,name:'test',isCompleted:false
    }]
  };


    const wrapper = shallow(<TodoItem {...props}/>);

  it("should render a todo list", () => {
    const wrapper = shallow(<TodoItem {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });


  it('should toggle todo completion',()=>{
    const wrapper = shallow(<TodoItem {...props}/>);
    expect(wrapper.find('.toggle')).toBeTruthy()

  })
});