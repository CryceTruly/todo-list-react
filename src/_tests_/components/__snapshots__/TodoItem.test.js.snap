// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp should render a todo list 1`] = `undefined`;
